types/json2csv/json2csv-tests.ts

```diff
@@ -36,11 +42,62 @@ const opts: json2csv.Options<Car> = {
     }]
 };
 
-const json2csvParser = new Parser<Car>(opts);
+const data = [{
+    car: 'VW Beetle',
+    price: 1395
+}, {
+    car: 'VW T1',
+    price: 1357
+}];
 
-parser.parse([]);
-const transform: Transform<ExampleObj> = new Transform<ExampleObj>({ quote: '' });
-const nodeTransform: NodeTransform = transform;
+// Test for Synchronous Parser
+try {
+    const parser = new Parser(opts);
+    const csv = parser.parse({ car: '', price: 1 });
+    console.log(csv);
+} catch (err) {
+    console.error(err);
+}
+
+// Test convenience method "parse"
+try {
+    const csv = parse(data, opts);
+    console.log(csv);
+} catch (err) {
+    console.error(err);
+}
+
+// Test for Asynchronous Parser
+const transformOpts = { highWaterMark: 8192 };
+
+const asyncParser = new AsyncParser(opts, transformOpts);
+
+let csv = '';
+asyncParser.processor
+    .on('data', chunk => (csv += chunk.toString()))
+    .on('end', () => console.log(csv))
+    .on('error', err => console.error(err));
+
+// Test for transform events
+asyncParser.transform
+    .on('header', header => console.log(header))
+    .on('line', line => console.log(line))
+    .on('error', err => console.log(err));
+
+asyncParser.input.push(data); // This data might come from an HTTP request, etc.
+asyncParser.input.push(null); // Sending `null` to a stream signal that no more data is expected and ends it.
+
+const input = createReadStream('/path/to/input', { encoding: 'utf8' });
+const output = createWriteStream('path/to/output', { encoding: 'utf8' });
+
+asyncParser.fromInput(input).toOutput(output).promise()
```

Comment:
I find this test a bit misleading.

You typically with either uses an output or use the promise, but not both.

Reply:
This example is just taken from the packages [README](https://github.com/zemirco/json2csv#json2csv-async-parser-streaming-api).
Thought the examples would make okayish test cases here.

@juanjoDiaz Can you provide a more fitting example?

Reply:
Probably I should review our own examples then 😅 

Typically you would do either `asyncParser.fromInput(input).toOutput(output)` or `asyncParser.fromInput(input).promise()`. The promise loads the whole resulting CSV in memory so it kind if defeat the purpose of streaming directly to an output (I'll revise the docs)

Reply:
Everything looks great so as soon as you split this into 2 examples, I'll approve the PR.

Thanks again!

Reply:
Done.